lib/run.js

```diff
@@ -158,10 +158,12 @@ setups.firefox = function (browser, options, callback) {
         return 'user_pref("' + name + '", ' + prefs[name] + ');';
     }).join('\n');
 
-    options.options = options.options.concat([
-        '--no-remote',
-        '-profile', profileDir
-    ]);
+    if (options.profile === undefined || options.profile !== null) {
```

Comment:
Maybe this should wrap all the rest of the code in this method? We're still creating the profile dir and populating lots of its configuration here even though (AFAICT) it won't be used anywhere.

That also highlights another point: if we don't use the generated profile, none of that other configuration (notably including the proxy settings) will do anything. 

That's not great! Are you sure that works for your use case? Even if so, this should probably throw an error if you try to set `options.proxy` or `options.prefs` with `profile: null` - in this current  code they're incompatible.

Reply:
That's up to you; I didn't have a look at all the other code as I was focused on solving my specific use case without breaking anyone else's current usage.

Ok let me make a new version with wrapping everything in the same null check and throw an error if proxy and prefs are set but profile is null...